src/nspawn/nspawn.c

```diff
@@ -1278,110 +1299,73 @@ static int parse_argv(int argc, char *argv[]) {
             !(arg_clone_ns_flags & CLONE_NEWUTS)) {
                 arg_register = false;
                 if (arg_start_mode != START_PID1)
-                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                               "--boot cannot be used without namespacing.");
+                        return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "--boot cannot be used without namespacing.");
         }
 
         if (arg_userns_mode == USER_NAMESPACE_PICK)
                 arg_userns_chown = true;
 
+        if (arg_start_mode == START_BOOT && arg_kill_signal <= 0)
+                arg_kill_signal = SIGRTMIN+3;
+
         if (arg_keep_unit && arg_register && cg_pid_get_owner_uid(0, NULL) >= 0)
                 /* Save the user from accidentally registering either user-$SESSION.scope or user@.service.
                  * The latter is not technically a user session, but we don't need to labour the point. */
-                return log_error_errno(SYNTHETIC_ERRNO(EINVAL),
-                                       "--keep-unit --register=yes may not be used when invoked from a user session.");
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "--keep-unit --register=yes may not be used when invoked from a user session.");
```

Comment:
github cuts off `oked from a user session.");` when I'm looking at this patch. Dunno, I understand why you like to keep everything on one line, but in practice it doesn't work out too well with the tools we use. It's also adds a lot of noise to the patch, making the interesting bits harder to see.